uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -545,13 +625,166 @@ define([
       /**
        * checks if all params are set to calculate balance
        *
-       * @param {Array} listOfDayTypes collection of available day types
+       * @param {Boolean} true if all present else false
        */
       function canCalculateChange() {
         return vm.leaveRequest.from_date && vm.leaveRequest.to_date &&
           vm.leaveRequest.from_date_type && vm.leaveRequest.to_date_type;
       }
 
+      /**
+       * checks if leaverequest is managed by given manager id
+       *
+       * @param {String} managerContactId
+       * @returns {Promise}
+       */
+      function isManagedBy(managerContactId) {
+        return vm.leaveRequest.roleOf({
+            id: managerContactId
+          })
+          .then(function (role) {
+            if (role === 'manager') {
+              vm.uiOptions.isManager = true;
+            } else {
+              vm.uiOptions.isManager = false;
+            }
+          });
+      }
+
+      /**
+       * initialize absence types
+       *
+       */
+      function initAbsenceType() {
+        if (canEdit()) {
+          vm.selectedAbsenceType = _.find(vm.absenceTypes, function (absenceType) {
+            return absenceType.id == vm.leaveRequest.type_id;
+          });
+        } else {
+          // Assign the first absence type to the leave request
+          vm.selectedAbsenceType = vm.absenceTypes[0];
+        }
+
+        vm.leaveRequest.type_id = vm.selectedAbsenceType.id;
+        // Init the `balance` object based on the first absence type
+        vm.balance.opening = vm.selectedAbsenceType.remainder;
+      }
+
+      /**
+       * initialize from and to dates, day types and contact
+       *
+       */
+      function initDates() {
+        initStatus();
+        initDayTypes();
+        initContact();
+
+        if (canEdit()) {
+          vm.uiOptions.fromDate = convertDateFormatFromServer(vm.leaveRequest.from_date);
+          vm.onDateChange(vm.uiOptions.fromDate, 'from');
+          vm.uiOptions.toDate = convertDateFormatFromServer(vm.leaveRequest.to_date);
+          vm.onDateChange(vm.uiOptions.toDate, 'to');
+        }
+      }
+
+      /**
+       * initialize day types
+       *
+       */
+      function initDayTypes() {
+        if (canEdit()) {
+          vm.uiOptions.selectedFromType = getDateTypeFromValue(vm.leaveRequest.from_date_type);
```

Comment:
I don't see the reason why you need `selectedFromType` and `selectedToType` to begin with. Why can't you plug `leaveRequest.from_day_type` and `leaveRequest.to_day_type` directly into the view? It would result in much less code